src/vs/workbench/contrib/terminal/browser/terminalActions.ts

```diff
@@ -142,6 +142,31 @@ export function registerTerminalActions() {
 			await accessor.get(ITerminalService).showProfileQuickPick('createInstance');
 		}
 	});
+	registerAction2(class extends Action2 {
+		constructor() {
+			super({
+				id: TERMINAL_COMMAND_ID.NEW_FROM_PROFILE,
+				title: { value: localize('workbench.action.terminal.newFromProfile', "Create New Integrated Terminal (From Profile)"), original: 'Create New Integrated Terminal (From Profile)' },
+				f1: true,
+				category,
+				description: {
+					description: 'workbench.action.terminal.newFromProfile',
+					args: [{
+						name: 'args',
+						schema: {
+							type: 'object'
```

Comment:
We can't re-register a command under the same ID, but we should be able to merge this with the above where we can optionally add `{ "profile": "name"}`

Reply:
I think we want the profile object right? not just the name?

Reply:
Since we already have to get the profile object itself here, I don't want to have to filter and find it via its profileName here

https://github.com/microsoft/vscode/blob/18c0f5ae8df2797adf445a0407247c0b349128e8/src/vs/workbench/contrib/terminal/browser/terminalActions.ts#L159-L161

Reply:
The name is a unique identifier, the code this calls into need to look up the profile to verify it's safe to launch as well (workspace permissions).